src/blog/2024/09/node-red-version-control-with-snapshots.md

```diff
@@ -0,0 +1,238 @@
+
+If you have multiple snapshots and want to compare them, follow these steps:
+
+![Image showing the 'Compare Snapshot' option in the menu](./images/compare-snapshot-option.png){data-zoomable}
+*Image showing the 'Compare Snapshot' option in the menu*
+
+1. Click on the three-dot icon next to the snapshot you want to compare, then choose **Compare Snapshot**.
+
+![Image showing the comparison window in FlowFuse, featuring a dropdown to select the instance to compare and a compare button.](./images/window-comparing-snapshots.png){data-zoomable}
+*Image showing the comparison window in FlowFuse, featuring a dropdown to select the instance to compare and a compare button*
+
+2. In the newly opened window, select another snapshot from the dropdown to compare it with the one you chose.
+
+![Image showing the comparison window for snapshots in FlowFuse, with differences highlighted in light purple color.](./images/window-comparing-snapshots-2.png){data-zoomable}
+*Image showing the comparison window for snapshots in FlowFuse.*
+
+
+3. The flows of both snapshots will be displayed side by side. At the top-right corner of the window, you will see two buttons, Use according to your preference:
+   - **Prev**: Navigate to the previous difference between the two snapshots.
+   - **Next**: Navigate to the next difference.
+
+This comparison feature helps you easily identify changes, making it simple to spot issues or track flow modifications between versions.
+
+### Downloading, Uploading, and Deploying Snapshots
+
+Created snapshots can be downloaded locally, providing you with a backup of your Node-RED flows. This backup can be uploaded to another instance or restored in case of emergencies.
+
+#### Downloading Snapshot:
+
+![Image showing the 'Download Snapshot' option in the menu](./images/download-snapshot.png){data-zoomable}
+*Image showing the 'Download Snapshot' option in the menu*
+
+1. Click on the **three-dot** icon next to the corresponding snapshot, then select **Download Snapshot** from the menu.
+
+![Image showing the prompt to download a snapshot in FlowFuse, featuring a highlighted key and the download button.](./images/download-snapshot-prompt.png){data-zoomable}
+*Image showing the prompt to download a snapshot in FlowFuse, featuring a highlighted key and the download button*
+
+2. A new prompt will open, displaying a **secret key**. This key is used to encrypt the configuration details of the snapshot. Copy the key and store it safely, as you will need it for restoring the snapshot.
+   - You can also generate a new secret key by clicking the **Random Secret** button on the right.
+3. After copying the secret key, click the **Download** button located at the bottom right of the prompt to download the snapshot.
+
+#### Uploading Snapshots:
+
+![Image showing the button to upload a snapshot in FlowFuse.](./images/button-upload-snapshot.png){data-zoomable}
+*Image showing the button to upload a snapshot in FlowFuse.*
```

Comment:
Should retake this screenshot with a file selected that actually has credentials - since it will show the "secret2 field (this screenshot does not contain the secret field)

ALSO NOTE:

Not all snapshots have credentials. They are only present when a flows node has a secret entered (like an MQTT Broker config node).  So when a snapshots flows do NOT have creds, the secret will NOT be asked and will not be shown on the download OR upload.